src/rails.js

```diff
@@ -309,24 +309,46 @@
 
     // Helper function which checks for blank inputs in a form that match the specified CSS selector
     blankInputs: function(form, specifiedSelector, nonBlank) {
-      var inputs = $(), input, valueToCheck,
-          selector = specifiedSelector || 'input,textarea',
-          allInputs = form.find(selector);
-
-      allInputs.each(function() {
+      var foundInputs = $(),
+        input,
+        valueToCheck,
+        radiosForNameWithNoneSelected,
+        radioName,
+        selector = specifiedSelector || 'input,textarea',
+        requiredInputs = form.find(selector),
+        checkedRadioButtonNames = {};
+
+      requiredInputs.each(function() {
         input = $(this);
         valueToCheck = input.is('input[type=checkbox],input[type=radio]') ? input.is(':checked') : !!input.val();
         if (valueToCheck === nonBlank) {
 
-          // Don't count unchecked required radio if other radio with same name is checked
-          if (input.is('input[type=radio]') && allInputs.filter('input[type=radio]:checked[name="' + input.attr('name') + '"]').length) {
-            return true; // Skip to next input
+          // Don't count unchecked required radio as blank if other radio with same name is checked,
+          // regardless of whether same-name radio input has required attribute or not. The spec
+          // states https://www.w3.org/TR/html5/forms.html#the-required-attribute
+          if (input.is('input[type=radio]')) {
+            radioName = input.attr('name');
+
+            // Skip if we've already seen the radio with this name.
+            if (checkedRadioButtonNames[radioName]) {
+              return true;
+            }
+
+            // If none checked
+            if (form.find('input[type=radio]:checked[name="' + radioName + '"]').length === 0) {
+              radiosForNameWithNoneSelected = form.find('input[type=radio][name="' + radioName + '"]');
+              foundInputs = foundInputs.add(radiosForNameWithNoneSelected);
```

Comment:
Just reading through the code, if you have a scenario where you have a required radio input with neither checked (i.e. the if-statement on line 338 above is true), does this end up adding the current `input` to `foundInputs` twice? It seems like it would, because `radiosForNameWithNoneSelected` would also include the current `input` radio, and thus get added to `foundInputs`, and then get added again below on line 348 (`foundInputs = foundInputs.add(input)`).

But I'm not certain how jQuery's `.add()` function works when you give it the same element twice. Does it skip it, or does it add the element to the collection twice? If it's the latter, seems like this could lead to some subtle bugs.